index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
>  (the `a` must be the same as well)

That's what I was getting at.  I see nothing in the implementation to enforce this.  Is there meant to be?

>  There is nothing specifying that the `a` is also a Semigroup in the implementation

I think you would need to enforce that for this to make any sense, or at least enforce a rule that the values have a `concat` function.  The question is whether you try to enforce something more stringent, namely that the `a`'s are of the same `Semigroup`.  If you do so, you get a useful `Semigroup` and with either of several possible extensions, you can get useful `Monoid`s out of it, as discussed on Gitter.  But I don't see anything here that enforces it.